tests/qadence/test_mitigation.py

```diff
@@ -123,6 +123,50 @@ def test_analog_zne_with_pulser(
     options = {"noise_probas": noise_probas}
     noise = Noise(protocol=noise_type, options=options)
     mitigation = Mitigations(protocol=Mitigations.ANALOG_ZNE)
-    noisy_expectation = model.expectation(noise=noise, mitigation=mitigation)
+    mitigated_expectation = model.expectation(noise=noise, mitigation=mitigation)
     exact_expectation = model.expectation()
-    assert torch.allclose(noisy_expectation, exact_expectation, atol=1.0e-2)
+    assert torch.allclose(mitigated_expectation, exact_expectation, atol=1.0e-2)
+
+
+@pytest.mark.parametrize(
+    "analog_block, observable, noise_probas, noise_type, param_values",
+    [
+        (
+            chain(AnalogRX("t"), AnalogRZ(pi)),
+            [Z(0) + Z(1)],
+            [0.1],
+            Noise.DEPOLARIZING,
+            {"t": torch.tensor([0.5])},
+        ),
+        # (
+        #     # Hardcoded time and angle for Bell state preparation.
+        #     chain(
+        #         entangle(383, qubit_support=(0, 1)),
+        #         RY(0, 3.0 * pi / 2.0),
+        #     ),
+        #     [hamiltonian_factory(2, detuning=Z)],
+        #     np.linspace(0.1, 0.5, 8),
+        #     Noise.DEPHASING,
+        # ),
+    ],
+)
+def test_analog_zne_with_pulse_stretching(
+    analog_block: AbstractBlock,
+    observable: AbstractBlock,
+    noise_probas: NDArray,
+    noise_type: str,
+    param_values: dict,
+) -> None:
+    circuit = QuantumCircuit(2, analog_block)
+    model = QuantumModel(
+        circuit=circuit, observable=observable, backend=BackendName.PULSER, diff_mode=DiffMode.GPSR
+    )
+    options = {"noise_probas": noise_probas}
+    noise = Noise(protocol=noise_type, options=options)
+    options = {"stretches": {"t": torch.tensor([1.5, 2.0, 2.5, 3.0])}}
```

Comment:
Does this mean that one needs to provide a dictionary with the stretched to be used for each variational parameter in the circuit? Cannot this be generalized/simplified in some way? Otherwise the interface looks a bit cumbersome.

Reply:
What are you thinking about it, for example?

Reply:
Well, I guess a simple dict with two entries should be enough:

```python
stretches = {"variables": ["t", <whatever-other-variable>], "stretches": torch.tensor([1.0, 2.0, 3.0])}
```

Reply:
And after discussing with @jpmoutinho, I will re-implement running noisy simulations in a way that makes better and more coherent use of the interface.